drivers/input/input_cf1133.c

```diff
@@ -0,0 +1,330 @@
+
+/** CF1133 configuration (DT). */
+struct cf1133_config {
+	/** I2C bus. */
+	struct i2c_dt_spec bus;
+	struct gpio_dt_spec reset_gpio;
+#ifdef CONFIG_INPUT_CF1133_INTERRUPT
+	/** Interrupt GPIO information. */
+	struct gpio_dt_spec int_gpio;
+#endif
+};
+
+/** CF1133 data. */
+struct cf1133_data {
+	/** Device pointer. */
+	const struct device *dev;
+	/** Work queue (for deferred read). */
+	struct k_work work;
+#ifdef CONFIG_INPUT_CF1133_INTERRUPT
+	/** Interrupt GPIO callback. */
+	struct gpio_callback int_gpio_cb;
+#else
+	/** Timer (polling mode). */
+	struct k_timer timer;
+#endif
+	/** Last pressed state. */
+	bool pressed_old;
+	bool pressed;
+
+	int resolution_x;
+	int resolution_y;
+	uint8_t touch_protocol_type;
+	uint8_t pixel_length;
+	uint8_t chip_id;
+};
+
+static int cf1133_get_chip_id(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int ret;
+	uint8_t buffer[3];
+	uint8_t num_x;
+	uint8_t num_y;
+
+	ret = i2c_burst_read_dt(&config->bus, CHIP_ID, buffer, sizeof(buffer));
+	if (ret < 0) {
+		LOG_ERR("Read burst failed: %d", ret);
+		return ret;
+	}
+	if (buffer[0] == 0) {
+		if (buffer[1] + buffer[2] > 32) {
+			data->chip_id = 2;
+		} else {
+			data->chip_id = 0;
+		}
+	} else {
+		data->chip_id = buffer[0];
+	}
+	num_x = buffer[1];
+	num_y = buffer[2];
+	LOG_DBG("Chip ID = %d, num_x = %d, num_y = %d", data->chip_id, num_x, num_y);
+
+	return 0;
+}
+
+static int cf1133_get_protocol_type(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int ret;
+	uint8_t buffer;
+	uint8_t sensing_mode;
+
+	if (data->chip_id <= 3) {
+		ret = i2c_reg_read_byte_dt(&config->bus, I2C_PROTOCOL, &buffer);
+		if (ret < 0) {
+			LOG_ERR("read i2c protocol failed: %d", ret);
+			return ret;
+		}
+		data->touch_protocol_type = FIELD_GET(I2C_PROTOCOL_BMSK, buffer);
+		LOG_DBG("i2c protocol = %d", data->touch_protocol_type);
+		sensing_mode = FIELD_GET(ONE_D_SENSING_CONTROL_BMSK << ONE_D_SENSING_CONTROL_SHFT,
+					 buffer);
+		LOG_DBG("sensing mode = %d", sensing_mode);
+	} else {
+		data->touch_protocol_type = SITRONIX_A_TYPE;
+		LOG_DBG("i2c protocol = %d", data->touch_protocol_type);
+
+		ret = i2c_reg_read_byte_dt(&config->bus, 0xf0, &buffer);
+		if (ret < 0) {
+			LOG_ERR("read sensing mode failed: (%d)", ret);
+			return ret;
+		}
+		sensing_mode = FIELD_GET(ONE_D_SENSING_CONTROL_BMSK, buffer);
+		LOG_DBG("sensing mode = %d", sensing_mode);
+	}
+	return 0;
+}
+
+static int cf1133_ts_init(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int r;
+
+	/* Get device status before use to do once */
+	r = cf1133_get_chip_id(data, config);
+	if (r < 0) {
+		LOG_ERR("Read chip id failed: %d", r);
+		return r;
+	}
+
+	r = cf1133_get_protocol_type(data, config);
+	if (r < 0) {
+		LOG_ERR("Read protocol failed: %d", r);
+		return r;
+	}
+
+	if (data->touch_protocol_type == SITRONIX_A_TYPE) {
+		data->pixel_length = PIXEL_DATA_LENGTH_A;
+		LOG_DBG("pixel length: %d", data->pixel_length);
```

Comment:
Nit- this log statement can be moved outside of the if statement

Reply:
Done